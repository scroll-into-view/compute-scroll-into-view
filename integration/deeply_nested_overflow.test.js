beforeAll(async () => {
  await page.goto('http://localhost:3000/integration/deeply_nested_overflow')
})

describe('scrollMode: if-needed', () => {
  const instructions = [
    [0, 'start', 'start'],
    [1, 'start', 'center'],
    [2, 'start', 'end'],
    [3, 'center', 'start'],
    [4, 'center', 'center'],
    [5, 'center', 'end'],
    [6, 'end', 'start'],
    [7, 'end', 'center'],
    [8, 'end', 'end'],
  ]
  instructions.forEach(([step, block, inline]) => {
    test(`test ${step}, block ${block}, inline ${inline}`, async () => {
      const actual = await page.evaluate(
        (i, block, inline) => {
          window.scrollTo(0, 0)
          return window
            .computeScrollIntoView(
              document.querySelectorAll('.scroll-tile')[i],
              {
                scrollMode: 'if-needed',
                block,
                inline,
              }
            )
            .map(window.mapActions)
        },
        step,
        block,
        inline
      )
      expect(actual).toHaveLength(3)
      expect(actual).toMatchSnapshot()
    })
  })
})
