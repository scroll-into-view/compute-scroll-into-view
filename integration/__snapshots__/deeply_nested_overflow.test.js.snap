// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scrollMode: if-needed test 0, block start, inline start 1`] = `
Array [
  Object {
    "el": "div.inner-scroll-container",
    "left": 55.546875,
    "top": 55.546875,
  },
  Object {
    "el": "div.outer-scroll-container",
    "left": 300,
    "top": 300,
  },
  Object {
    "el": "html",
    "left": 1000,
    "top": 1200,
  },
]
`;

exports[`scrollMode: if-needed test 1, block start, inline center 1`] = `
Array [
  Object {
    "el": "div.inner-scroll-container",
    "left": 99.96875,
    "top": 55.546875,
  },
  Object {
    "el": "div.outer-scroll-container",
    "left": 200,
    "top": 300,
  },
  Object {
    "el": "html",
    "left": 800,
    "top": 1200,
  },
]
`;

exports[`scrollMode: if-needed test 2, block start, inline end 1`] = `
Array [
  Object {
    "el": "div.inner-scroll-container",
    "left": 144.390625,
    "top": 55.546875,
  },
  Object {
    "el": "div.outer-scroll-container",
    "left": 100,
    "top": 300,
  },
  Object {
    "el": "html",
    "left": 600,
    "top": 1200,
  },
]
`;

exports[`scrollMode: if-needed test 3, block center, inline start 1`] = `
Array [
  Object {
    "el": "div.inner-scroll-container",
    "left": 55.546875,
    "top": 99.984375,
  },
  Object {
    "el": "div.outer-scroll-container",
    "left": 300,
    "top": 200,
  },
  Object {
    "el": "html",
    "left": 1000,
    "top": 1100,
  },
]
`;

exports[`scrollMode: if-needed test 4, block center, inline center 1`] = `
Array [
  Object {
    "el": "div.inner-scroll-container",
    "left": 99.96875,
    "top": 99.984375,
  },
  Object {
    "el": "div.outer-scroll-container",
    "left": 200,
    "top": 200,
  },
  Object {
    "el": "html",
    "left": 800,
    "top": 1100,
  },
]
`;

exports[`scrollMode: if-needed test 5, block center, inline end 1`] = `
Array [
  Object {
    "el": "div.inner-scroll-container",
    "left": 144.390625,
    "top": 99.984375,
  },
  Object {
    "el": "div.outer-scroll-container",
    "left": 100,
    "top": 200,
  },
  Object {
    "el": "html",
    "left": 600,
    "top": 1100,
  },
]
`;

exports[`scrollMode: if-needed test 6, block end, inline start 1`] = `
Array [
  Object {
    "el": "div.inner-scroll-container",
    "left": 55.546875,
    "top": 144.421875,
  },
  Object {
    "el": "div.outer-scroll-container",
    "left": 300,
    "top": 100,
  },
  Object {
    "el": "html",
    "left": 1000,
    "top": 1000,
  },
]
`;

exports[`scrollMode: if-needed test 7, block end, inline center 1`] = `
Array [
  Object {
    "el": "div.inner-scroll-container",
    "left": 99.96875,
    "top": 144.421875,
  },
  Object {
    "el": "div.outer-scroll-container",
    "left": 200,
    "top": 100,
  },
  Object {
    "el": "html",
    "left": 800,
    "top": 1000,
  },
]
`;

exports[`scrollMode: if-needed test 8, block end, inline end 1`] = `
Array [
  Object {
    "el": "div.inner-scroll-container",
    "left": 144.390625,
    "top": 144.421875,
  },
  Object {
    "el": "div.outer-scroll-container",
    "left": 100,
    "top": 100,
  },
  Object {
    "el": "html",
    "left": 600,
    "top": 1000,
  },
]
`;
